### PR TITLE
[SDK-3285] Add Python Web App Interactive QuickStart

### PR DESCRIPTION
**Pull request by adamjmcgrath**
image

### CODE REVIEW

**Review by frederikprijck — COMMENTED** [review]
Should this need a checkpoint section as mentioned in https://oktawiki.atlassian.net/wiki/spaces/DXSDK/pages/2540963444/How+to+Migrate+a+Quickstart+to+the+New+Design#Checkpoints ?

Using a checkpoint is the new way to gather feedback from the user regarding the quickstart.

**Comment by adamjmcgrath** [comment]
Ah, yep, good point @frederikprijck - will add one to this (and the other PRs I have open) 👍

**Review by frederikprijck — DISMISSED** [review]

File: articles/quickstart/webapp/python/interactive.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+::::checkpoint
+:::checkpoint-default
+Your application should be running, when you visit [http://localhost:3000](http://localhost:3000) you should see a login button that takes you to Auth0 to login, then back to your application to see your profile information.
+:::
+
+:::checkpoint-failure
+Sorry about that. Here's a couple things to double check:
+* Make sure there are no errors in the console.
+* Make sure the domain and client ID imported correctly.
```
Comment (stevehobbsdev) [inline articles/quickstart/webapp/python/interactive.md]: Style guide (word list): Client ID should be capitalized

```suggestion
* Make sure the domain and Client ID imported correctly.
```

**Review by stevehobbsdev — CHANGES_REQUESTED** [review]

**Review by frederikprijck — DISMISSED** [review]

**Review by stevehobbsdev — DISMISSED** [review]

File: articles/quickstart/webapp/python/interactive.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  - flask
+github:
+  path: 01-Login
+---
+
+<!-- markdownlint-disable MD025 MD034 -->
+
+# Add login to your Python Flask app
+
+Auth0 allows you to quickly add authentication and gain access to user profile information in your application. This guide demonstrates how to integrate Auth0 with a Python [Flask](https://flask.palletsprojects.com) application using the [Authlib](https://authlib.org/) SDK.
```
Comment (tr0njavolta) [inline articles/quickstart/webapp/python/interactive.md]:
```suggestion
Auth0 allows you to add authentication and gain access to user profile information in your application. This guide demonstrates how to integrate Auth0 with a Python [Flask](https://flask.palletsprojects.com) application using the [Authlib](https://authlib.org/) SDK.
```

File: articles/quickstart/webapp/python/interactive.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+<%= include('../../_includes/_configure_auth0_interactive', {
+callback: 'http://localhost:3000/callback',
+returnTo: 'http://localhost:3000'
+}) %>
+
+## Install dependencies
+
+Begin by creating a `requirements.txt` file in your project directory:
```
Comment (tr0njavolta) [inline articles/quickstart/webapp/python/interactive.md]:
```suggestion
Create a `requirements.txt` file in your project directory:
```

File: articles/quickstart/webapp/python/interactive.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+```python
+# 📁 requirements.txt -----
+
+flask>=2.0.3
+python-dotenv>=0.19.2
+authlib>=1.0
+requests>=2.27.1
+```
+
+You should now run `pip install -r requirements.txt` from your shell to make these dependencies available to your project.
```
Comment (tr0njavolta) [inline articles/quickstart/webapp/python/interactive.md]:
```suggestion
Run the following command from your shell to enable these dependencies in your project:

```sh
pip install -r requirements.txt
```
```

File: articles/quickstart/webapp/python/interactive.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+```ini
+# 📁 .env -----
+
+AUTH0_CLIENT_ID=${account.clientId}
+AUTH0_CLIENT_SECRET=${account.clientSecret}
+AUTH0_DOMAIN=${account.namespace}
+APP_SECRET_KEY=
+```
+
+- Generate a suitable string for `APP_SECRET_KEY` using `openssl rand -hex 32` from your shell.
```
Comment (tr0njavolta) [inline articles/quickstart/webapp/python/interactive.md]:
```suggestion
- Generate a string for `APP_SECRET_KEY` using `openssl rand -hex 32` from your shell.
```

File: articles/quickstart/webapp/python/interactive.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+## Setup your application {{{ data-action=code data-code="server.py#1:27" }}}
+
+Now you're ready to start writing your application. Create a `server.py` file in your project directory - this file will hold all of your application logic.
```
Comment (tr0njavolta) [inline articles/quickstart/webapp/python/interactive.md]:
```suggestion
Next, set up your application. Create a `server.py` file in your project directory - this file will contain your application logic.
```

File: articles/quickstart/webapp/python/interactive.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+Begin by importing all the libraries your application will be making use of.
```
Comment (tr0njavolta) [inline articles/quickstart/webapp/python/interactive.md]:
```suggestion
Import all the libraries your application needs.
```

File: articles/quickstart/webapp/python/interactive.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+
+Then load the configuration `.env` file you made in the previous step.
```
Comment (tr0njavolta) [inline articles/quickstart/webapp/python/interactive.md]:
```suggestion
Load the configuration `.env` file you made in the previous step.
```

File: articles/quickstart/webapp/python/interactive.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+Then configure Authlib to handle your application's authentication with Auth0. You can learn more about the configuration options available for Authlib's OAuth `register()` method from [their documentation.](https://docs.authlib.org/en/latest/client/frameworks.html#using-oauth-2-0-to-log-in)
```
Comment (tr0njavolta) [inline articles/quickstart/webapp/python/interactive.md]:
```suggestion
Configure Authlib to handle your application's authentication with Auth0. To learn more about the configuration options available for Authlib's OAuth `register()` method from [their documentation.](https://docs.authlib.org/en/latest/client/frameworks.html#using-oauth-2-0-to-log-in)
```

File: articles/quickstart/webapp/python/interactive.md [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+
+## Setup your routes {{{ data-action=code data-code="server.py#30:59" }}}
+
+For this demonstration, we'll be adding 4 routes for your application: your login, callback, logout and home routes.
```
Comment (tr0njavolta) [inline articles/quickstart/webapp/python/interactive.md]:
```suggestion
In this example, you will add four routes to the application: login, callback, logout, and home.
```

File: articles/quickstart/webapp/python/interactive.md [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+
+When visitors to your app visit the `/login` route, they'll be redirected to Auth0 to begin the authentication flow.
```
Comment (tr0njavolta) [inline articles/quickstart/webapp/python/interactive.md]:
```suggestion
When visitors to your app visit the `/login` route, your application will route them to the Auth0 login page.
```

File: articles/quickstart/webapp/python/interactive.md [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+
+After your users finish logging in with Auth0, they'll be returned to your application at the `/callback` route. This route is responsible for actually saving the session for the user, so when they visit again later, they won't have to sign back in all over again.
```
Comment (tr0njavolta) [inline articles/quickstart/webapp/python/interactive.md]:
```suggestion
After your users log in with Auth0, your application will route them to the `/callback` route. This route saves the session for the user and bypasses the need for them to login again when they return.
```

File: articles/quickstart/webapp/python/interactive.md [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+
+As you might expect, the `/logout` route handles signing a user out from your application. It will clear the user's session in your app, and briefly redirect to Auth0's logout endpoint to ensure their session is completely clear, before they are returned to your home route.
```
Comment (tr0njavolta) [inline articles/quickstart/webapp/python/interactive.md]:
```suggestion
The `/logout` route signs users out from your application. This route clears the user session in your app and redirects to the Auth0 logout endpoint to ensure the session is no longer saved. Then, the application redirects the user to your home route.
```

File: articles/quickstart/webapp/python/interactive.md [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+
+Last but not least, your `/` home route will serve as a place to either render an authenticated user's details, or offer to allow visitors to sign in.
```
Comment (tr0njavolta) [inline articles/quickstart/webapp/python/interactive.md]:
```suggestion
Your `/ ` home route either renders an authenticated user's details or  allows visitors to sign in.
```

File: articles/quickstart/webapp/python/interactive.md [hunk 15]
```diff
@@ -0,0 +1,110 @@
...
+
+## Add templates {{{ data-action=code data-code="templates/home.html" }}}
+
+Now we just need to create the simple template file used in the home route (during `render_template()` calls).
+
+Create a new sub-directory in your project folder named `templates`, and create `home.html`. You can paste the content from the right into that file.
```
Comment (tr0njavolta) [inline articles/quickstart/webapp/python/interactive.md]:
```suggestion
Next, create the template file used in the home route (during `render_template()` calls).

Create a new sub-directory in your project folder named `templates`, and create `home.html`  in the directory. Paste the content from the right into that file.
```

File: articles/quickstart/webapp/python/interactive.md [hunk 16]
```diff
@@ -0,0 +1,110 @@
...
+
+## Run your application
+
+You're ready to run your application! From your project directory, open a shell and use:
```
Comment (tr0njavolta) [inline articles/quickstart/webapp/python/interactive.md]:
```suggestion
To run your application, navigate to the root of your project directory and open a terminal. Run the following command:
```

File: articles/quickstart/webapp/python/interactive.md [hunk 17]
```diff
@@ -0,0 +1,110 @@
...
+
+```sh
+python3 server.py
+```
+
+::::checkpoint
+:::checkpoint-default
+Your application should be running, when you visit [http://localhost:3000](http://localhost:3000) you should see a login button that takes you to Auth0 to login, then back to your application to see your profile information.
```
Comment (tr0njavolta) [inline articles/quickstart/webapp/python/interactive.md]:
```suggestion
Visit [http://localhost:3000](http://localhost:3000) to verify. You should find a login button routing to Auth0 for login, then back to your application to see your profile information.
```

File: articles/quickstart/webapp/python/interactive.md [hunk 18]
```diff
@@ -0,0 +1,110 @@
...
+:::
+
+:::checkpoint-failure
+Sorry about that. Here's a couple things to double check:
+* Make sure there are no errors in the console.
+* Make sure the domain and Client ID imported correctly.
+* Make sure your tenant is correctly configured.
+
```
Comment (tr0njavolta) [inline articles/quickstart/webapp/python/interactive.md]:
```suggestion
If your application did not start successfully:
* Verify any errors in the console.
* Verify the domain and Client ID imported correctly.
* Verify your tenant configuration.

```

**Review by tr0njavolta — COMMENTED** [review]
Some comments for clarity and style guide. Let me know if these can be integrated and I'll approve.

**Comment by frederikprijck** [comment]
Thanks @tr0njavolta , I updated the PR to incorporate the suggestions.

**Review by tr0njavolta — APPROVED** [review]

**Pull request merged**